[CI] CrossClusterAsyncQueryStopIT testStopQueryLocal failing

**Build Scans:**
- [elasticsearch-intake #25576 / part3](https://gradle-enterprise.elastic.co/s/wt22s2jq33ud4)
- [elasticsearch-periodic #9376 / release-tests](https://gradle-enterprise.elastic.co/s/7icyhnxodaxre)
- [elasticsearch-periodic #9369 / release-tests](https://gradle-enterprise.elastic.co/s/xhva7zbxxfvz6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT.testStopQuery" -Dtests.seed=F3E498874ABA4D1C -Dtests.locale=de-CH -Dtests.timezone=America/Yellowknife -Druntime.java=24
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testStopQueryLocal'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request breaker not reset to 0 on node: main-cluster-0
Expected: <0L>
     but: was <32L>
```

**Issue Reasons:**
- [9.1] 2 consecutive failures in step release-tests
- [9.1] 3 failures in test testStopQueryLocal (4.2% fail rate in 71 executions)
- [9.1] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [9.1] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -268,9 +268,6 @@ tests:
 - class: org.elasticsearch.smoketest.SmokeTestMonitoringWithSecurityIT
   method: testHTTPExporterWithSSL
   issue: https://github.com/elastic/elasticsearch/issues/122220
-- class: org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT
-  method: testStopQueryLocal
-  issue: https://github.com/elastic/elasticsearch/issues/121672
 - class: org.elasticsearch.blocks.SimpleBlocksIT
   method: testConcurrentAddBlock
   issue: https://github.com/elastic/elasticsearch/issues/122324

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterAsyncQueryStopIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterAsyncQueryStopIT.java
@@ -156,7 +156,7 @@ public class CrossClusterAsyncQueryStopIT extends AbstractCrossClusterTestCase {
                 try (EsqlQueryResponse asyncResponse = getAsyncResponse(client(), asyncExecutionId)) {
                     EsqlExecutionInfo executionInfo = asyncResponse.getExecutionInfo();
                     assertNotNull(executionInfo);
-                    assertThat(executionInfo.isPartial(), is(true));
+                    assertThat(executionInfo.isStopped(), is(true));
                 }
             });
             // allow local query to proceed

